aea/agent_loop.py

```diff
@@ -279,7 +291,7 @@ async def _task_process_new_skill_components(self) -> None:
             # TODO: better handling internal messages for skills internal updates
             self._agent.filter._handle_new_behaviours()  # pylint: disable=protected-access # TODO: refactoring!
             self._agent.filter._handle_new_handlers()  # pylint: disable=protected-access # TODO: refactoring!
```

Comment:
I think we should call `self._agent.update` here. Then `update` in `aea.py` needs to be changed to:
```
    def update(self) -> None:
        """
        Update the current state of the agent.

        Handles the internal messages from the skills to the decision maker.

        :return None
        """
        self.filter.update_behaviours_and_handlers()
```
and in filter; we need to change `handle_internal_messages` to 
```
    def update_ behaviours_and_handlers(self) -> None:
        """
        Update the behaviours and handlers.

        :return: None
        """
        # get new behaviours and handlers from the agent skills
        self._handle_new_behaviours()
        self._handle_new_handlers()
```

Reply:
will do as a part of another pr.
`update` quite tricky at the moment and requires a better overview

Reply:
The part I am suggesting is not tricky; you can implement it how I outlined it.